### ISSUE TITLE
string property after decompress became array

### ISSUE THREAD

**Issue by Cherviakov** [beenotung/compress-json#2]
Tried to use this package for app I working on, have big complex nested structures json, one important property was string, but after decompress became array. Will try a bit later to make reproducable example as can't really share original json.

**Comment by beenotung** [comment]
Look forward to an example reproducing this bug

**Comment by beenotung** [comment]
@Cherviakov wish you could give an example reproducing the bug, then we can fix it

**Comment by Cherviakov** [comment]
Here it is, though now error is different
```js
{
        "json": {
                "versions": [
                        {
                                "things": [
                                        {
                                                "ID": 1,
                                                "GUID": "e96bd89e-86eb-42da-99c9-345adb3a89d6",
                                                "Name": "Test",
                                                "Description": "",
                                                "Subthings": [
                                                        {
                                                                "ID": 7,
                                                                "GUID": "10666c0a-12a5-44e9-860c-684b7f1de6ab",
                                                                "Name": "Start",
                                                                "Triangles": {
                                                                        "Triangles": [
                                                                                {
                                                                                        "Name": "Triangle-01"
                                                                                }
                                                                        ]
                                                                }
                                                        }
                                                ]
                                        }
                                ]
                        }
                ]
        }
}
```

Here is also code snippet I used to check what is wrong:
```js
const { compress, decompress } = require('compress-json');
const test = require('./Test.json');

const isEqual = (json1, json2) => {
  return JSON.stringify(json1) === JSON.stringify(json2);
};

const com = compress(test);
const decom = decompress(com);
console.log(isEqual(test, decom));
console.log(isEqual(test.json.versions[0].things[0].Subthings[0].Triangles.Triangles, decom.json.versions[0].things[0].Subthings[0].Triangles.Triangles));
console.log(test.json.versions[0].things[0].Subthings[0].Triangles);
console.log(decom.json.versions[0].things[0].Subthings[0].Triangles);
```
Output I got
```sh
false
false
{ Triangles: [ { Name: 'Triangle-01' } ] }
{ T: [ { N: 'Triangle-01' } ] }
```

Additional info:
os: Linux-Mint-18 4.15.0-129-generic #132~16.04.1-Ubuntu SMP Wed Dec 16 06:46:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
node: v15.0.1
npm: 7.0.3

**Comment by beenotung** [comment]
Thanks for the sample data. I can reproduce the problem, hopefully will resolve it over this weekend.

**Comment by beenotung** [comment]
The bug when running above sample data is fixed and the new version is published as v1.0.2 on npm.

However, the original reported bug 'string property after decompress became array' is not reproduced yet.

**Comment by beenotung** [comment]
The reported bug(s) should have been fixed in compress-json@1.0.3.
@Cherviakov, might you confirm that with your data set? Thanks.

**Comment by Cherviakov** [comment]
Tested, can't reproduce nor last problem neither original one. Thanks.
